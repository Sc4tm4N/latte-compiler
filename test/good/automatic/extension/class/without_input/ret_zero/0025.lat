class a {
  int i;
  string s;
  boolean b;
  editor e;

  void doHast() {
    // what can go wrong?
    e.iWillEditYou(self);
  }
}

class editor {

  void iWillEditYou(a a) {
    a.i = 100000;
    a.s = "edited";
    a.b = !a.b;
  }

  void iWillEditYouDifferently(a a) {
    a.i = 666;
    a.s = "wasted";
    a.b = !a.b;
  }
}

int main() {
  editor editor = new editor;
  a a = new a;
  a.i = 5;
  a.s = "No... mercy!";
  a.b = true;
  a.e = editor;

  printInt(a.i);
  printString(a.s);
  printBoolean(a.b);

  a.doHast();

  printInt(a.i);
  printString(a.s);
  printBoolean(a.b);

  a.e.iWillEditYouDifferently(a);

  printInt(a.i);
  printString(a.s);
  printBoolean(a.b);

  return 0;
}

void printBoolean(boolean b) {
  if(b) {
    printString("true");
  } else {
    printString("false");
  }
}