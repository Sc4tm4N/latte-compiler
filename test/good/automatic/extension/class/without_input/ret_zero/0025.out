5
No... mercy!
true
100000
edited
false
666
wasted
true
